docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+### Preparing your JavaScript codebase for the new React Native Renderer (Fabric)
+
+The new renderer also known as Fabric doesn’t use the UIManager so direct calls to UIManager will need to be migrated. Historically, calls to UIManager had some pretty complicated patterns. Fortunately, we’ve created new APIs that are much cleaner. These new APIs are forwards compatible with Fabric so you can migrate your code today and they will work properly when you turn on Fabric!
+
+Fabric will be providing new type safe JS APIs that are much more ergonomic than some of the patterns we've seen in product code today. These APIs require references to the underlying component, no longer using the result of `findNodeHandle`. `findNodeHandle` is used to search the tree for a native component given a class instance. This was breaking the React abstraction model. `findNodeHandle` won’t be compatible with React concurrent mode once we are ready to roll that out. Deprecation of `findNodeHandle` in React Native is similar to the [deprecation of `findDOMNode` in React DOM](https://reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage).
+
+While we know that all deprecations are a hassle, this guide is intended to help people update components as smoothly as possible. Here are the steps you need to take to get your JS codebase ready for Fabric
+
+1. Migrating findNodeHandle / getting a HostComponent
+2. Migrating `.measure*(`
+3. Migrating off `setNativeProps`
+4. Move the call to `requireNativeComponent` to a separate file
+5. Migrating off `dispatchViewManagerCommand`
+6. Using `codegenNativeComponent`
+
+#### Migrating `findNodeHandle` / getting a `HostComponent`
+
+<!-- alex ignore host -->
+
+Much of the migration work requires a HostComponent ref to access certain APIs that are only attached to host components (like View, Text, or ScrollView). HostComponents are the return value of calls to `requireNativeComponent`. `findNodeHandle` tunnels through multiple levels of component hierarchy to find the nearest native component.
+
+As a concrete example, this code uses `findNodeHandle` to tunnel from `MyComponent` through to the `View` rendered by `MyJSComponent`.
+
+```javascript
+class MyComponent extends React.Component<Props> {
+  _ref: ?React.ElementRef<typeof MyJSComponent>;
+
+  render() {
+    return <MyJSComponent ref={this._captureRef} onSubmit={this._onSubmit} />
+  }
+
+  _captureRef: (ref) => {
+    this._ref = ref;
+  }
+
+  _onSubmit: () => {
+    const nodeHandle = findNodeHandle(this._ref);
+
+    if (nodeHandle) {
+      UIManager.measure(nodeHandle, () => {});
+    }
+  }
+}
+
+function MyJSComponent(props) {
+  return (
+    <View>
+      <SubmitButton onSubmit={props.onSubmit} />
+    </View>
+  );
+}
+```
+
+We can’t convert this call to `this._ref.measure` because `this._ref` is an instance to `MyJSComponent`, which is not a HostComponent and thus does not have a `measure` function.
```

Comment:
I don't believe this claim is correct. I believe the code _above_ does not work (even in Paper) because you can't have a `ref` to a function component. (That's not related to Fabric and is how React works in general.) The only way this could've worked in the past is if `MyJSComponent` is a class or if `MyJSComponent` is a `forwardRef` wrapper.

Concretely, I would suggest converting the child component to a class for this example to make sense. If it was a function you _already_ couldn't `findNodeHandle` on it anyway so you didn't have this problem.